### PR TITLE
[system-probe] Retry failed offset guessing

### PR DESCRIPTION
**Pull request by ISauve**
### What does this PR do?

Adds logic to retry offset guessing if it fails (up to 5 times max).

### Motivation

Customers have reported that offset guessing is intermittently failing on their system-probe containers, causing them to have trouble setting up NPM.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details that should be tested during the QA.

### CODE REVIEW

**Comment by hmahmood** [comment]
Do we know what error is being reported?

**Comment by ISauve** [comment]
@hmahmood Yes, it's the following:
```
new module `network_tracer` error: error guessing offsets: invalid guessing state while guessing source address, got checking expected checked. tcp_getsockopt kprobe not executed
```

**Review by hmahmood — APPROVED** [review]

**Comment by brycekahle** [comment]
@ISauve does it work on restart? I wouldn't expect retrying to have different results.

**Comment by ISauve** [comment]
@brycekahle According to what 1 customer was reporting, yes it works on restart. The other report has not confirmed yet whether the issue is solved on restart or not

File: pkg/network/tracer/tracer.go [hunk 1]
```diff
@@ -182,7 +182,13 @@ func NewTracer(config *config.Config) (*Tracer, error) {
 		}
 		defer offsetBuf.Close()
 
-		mgrOptions.ConstantEditors, err = runOffsetGuessing(config, offsetBuf)
+		// Offset guessing has been flaky for some customers, so if it fails we'll retry it up to 5 times
+		for i := 0; i < 5; i++ {
+			mgrOptions.ConstantEditors, err = runOffsetGuessing(config, offsetBuf)
+			if err == nil {
```
Comment (mbakht) [inline pkg/network/tracer/tracer.go:188]: I wonder if it would make sense to do a `sleep` between retries.

**Review by mbakht — APPROVED** [review]
Please add the `skip QA` label if no QA is needed.

File: pkg/network/tracer/tracer.go (hunk 1 above)
Comment (ISauve) [reply pkg/network/tracer/tracer.go:188]: 🤔  I don't think it could hurt, I'll add that!

**Pull request merged**